icepyx/core/granules.py

```diff
@@ -261,13 +277,13 @@ def get_avail(self, CMRparams, reqparams, cloud=False):
     # DevGoal: add kwargs to allow subsetting and more control over request options.
     def place_order(
         self,
-        CMRparams,
-        reqparams,
+        CMRparams: CMRParams,
+        reqparams: EGIRequiredParamsDownload,
         subsetparams,
```

Comment:
```suggestion
        subsetparams: Union[EGIRequiredParamsSubset, dict[Never, Never]],
```

I might have the syntax wrong. In some cases we specify the type from `types.py`, and in others from `APIformatting.py` (e.g. in `query.py`).

Reply:
`dict[Never, Never]` means an empty dict, and this change doesn't pass type checking. In what case would `reqparams={}` be passed?

Reply:
Woops, I meant `subsetparams` :)